### PR TITLE
DAT-55993 - DQL Editor changes and bug fixes

### PR DESCRIPTION
**Pull request by makc-dataloop**
@guyDataloop as discussed on slack:

Screenshot 2023-12-23 at 10 37 34 AM

### CODE REVIEW

**Comment by fadiDL** [comment]
@makc-dataloop As this is a public repo in the future please add basic information to the title of the pull request and link it to an active issue. if there is no need to link to an issue please provide in the description the related jira ticket.

File: src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchJsonEditorDialog.vue [hunk 1]
```diff
@@ -21,8 +21,21 @@
                     >
                         <dl-select
                             :model-value="selectedOption"
-                            width="200px"
-                            :options="selectOptions"
+                            width="300px"
+                            :options="
+                                selectOptions.map((option) => {
```
Comment (guyDataloop) [inline src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchJsonEditorDialog.vue:26]: Use regular `span` with `v-for`, avoid injecting html

File: src/components/compound/DlSelect/DlSelect.vue [hunk 2]
```diff
@@ -119,9 +119,8 @@
                                 (!multiselect && searchable && !isExpanded)
                         "
                         class="selected-label"
-                    >
-                        {{ getLabel(selectedOption) }}
-                    </span>
+                        v-html="getLabel(selectedOption)"
```
Comment (guyDataloop) [inline src/components/compound/DlSelect/DlSelect.vue:122]: Same here, why do oyu need to inject html ?

**Review by guyDataloop — CHANGES_REQUESTED** [review]

**Review by fadiDL — CHANGES_REQUESTED** [review]

File: src/components/compound/DlSelect/DlSelect.vue (hunk 2 above)
Comment (makc-dataloop) [reply src/components/compound/DlSelect/DlSelect.vue:122]: @guyDataloop okay, copying from slack:
> I’ve found that DlSelect drop down menu does treat option.label as html (for searchable mode support), so just by doing the same while displaying selected option we can provide custom html for options display everywhere in DlSelect. this seems like the least invasive fix

File: src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchJsonEditorDialog.vue [hunk 3]
```diff
@@ -21,8 +21,21 @@
...
+                                    return {
+                                        ...option,
+                                        label: `<span style='
```
[the remark this answers is not in the document]
Comment (makc-dataloop) [reply src/components/compound/DlSearches/DlSmartSearch/components/DlSmartSearchJsonEditorDialog.vue:29]: @fadiDL does not that slot sit in drop-down menu only? that would not do anything to selected option display.

**Comment by makc-dataloop** [comment]
@fadiDL updated with the requested changes ^

**Review by fadiDL — APPROVED** [review]

**Review by fadiDL — APPROVED** [review]

**Pull request merged**